ssl/ssl_lib.c

```diff
@@ -25,27 +25,84 @@
 
 const char SSL_version_str[] = OPENSSL_VERSION_TEXT;
 
+static int ssl_undefined_function_1(SSL *ssl, SSL3_RECORD *r, size_t s, int t)
```

Comment:
Don'tcha wish C had the "unnamed param means unused" thing from C++? :)

Reply:
yes.

Reply:
Is a C++ build possible?  Keeping the external interfaces as C linkage of course.

Reply:
Could these be defined by disabling the unused argument warnings explicitly?

Reply:
I never tried, but there are some quite subtle differences between C and C++,
where is a warning: -Wc++-compat
"Warn about C constructs that are not in the common subset of C and C++"

Reply:
> Could these be defined by disabling the unused argument warnings explicitly?

Sorry, what ?

Reply:
It is technically possible to compile (portions of) the library internals as C++ but only expose C-compatible linkage public functions.  The details are fairly compiler/linker-specific and may still result in the need for a C++ runtime ... requiring either a C++ compiler or runtime would probably shrink the list of platforms we could build on by an amount that would generate enough complaints as to make it infeasible, as convenient as it would be for us the authors.
(BTW BoringSSL is well into a C++ conversion that IIRC started in this vein, but may eventually end up exposing C++ APIs; I wasn't following it too closely.)

Reply:
I don't think it's feasible to do the transition to compile with C++ before 1.2.0.  I'd rather not see that happens at all if we can help it.  We claim to support ANSI C at least when it comes to language syntax and semantics, that's C89 or C90 depending on who you ask (and the difference between the two is very small).

*If* we would ever go to C++, I'd rather we do that fully, i.e. that we make good use of that language.  That would, however, become an entirely new library, not OpenSSL any more...

Reply:
I think the question is, can you build openssl with a C++ compiler?  No, that is not currently supported.  Can you write a C++ program that uses openssl headers and libraries?  Yes.